src/overlay/OverlayManagerImpl.h

```diff
@@ -164,6 +165,8 @@ class OverlayManagerImpl : public OverlayManager
     void start() override;
     void shutdown() override;
 
+    void updateSnapshots() override;
```

Comment:
I don't think this is needed: the underlying bucket snapshot in LedgerSnapshot is managed by LedgerManager already, so if you create a new LedgerSnapshot every time instead of caching it, you'll automatically get the most recent version of the ledger. No need for snapshot mutex either, since bucketlist are already refreshed in a thread-safe manner.